exporter/zipkinexporter/zipkin.go

```diff
@@ -160,52 +188,38 @@ func extractStringAttribute(
 	return value, ok
 }
 
-func (ze *zipkinExporter) Name() string {
-	return ze.defaultServiceName
-}
-
-func (ze *zipkinExporter) Start(host exporter.Host) error {
-	return nil
-}
-
-func (ze *zipkinExporter) Shutdown() error {
-	ze.mu.Lock()
-	defer ze.mu.Unlock()
-
-	return ze.reporter.Close()
-}
-
-func (ze *zipkinExporter) ConsumeTraceData(ctx context.Context, td consumerdata.TraceData) (zerr error) {
-	ctx, span := trace.StartSpan(ctx,
-		"opencensus.service.exporter.zipkin.ExportTrace",
-		trace.WithSampler(trace.NeverSample()))
+func (ze *zipkinExporter) PushTraceData(ctx context.Context, td consumerdata.TraceData) (droppedSpans int, err error) {
 
-	defer func() {
-		if zerr != nil && span.IsRecordingEvents() {
-			span.SetStatus(trace.Status{Code: trace.StatusCodeInternal, Message: zerr.Error()})
-		}
-		span.End()
-	}()
-
-	goodSpans := 0
+	tbatch := []*zipkinmodel.SpanModel{}
 	for _, span := range td.Spans {
 		sd, err := spandatatranslator.ProtoSpanToOCSpanData(span)
 		if err != nil {
-			return consumererror.Permanent(err)
+			return len(td.Spans), consumererror.Permanent(err)
 		}
 		zs := ze.zipkinSpan(td.Node, sd)
-		// ze.reporter can get closed in the midst of a Send
-		// so avoid a read/write during that mutation.
-		ze.mu.Lock()
-		ze.reporter.Send(zs)
-		ze.mu.Unlock()
-		goodSpans++
+		tbatch = append(tbatch, &zs)
 	}
 
-	// And finally record metrics on the number of exported spans.
-	observability.RecordMetricsForTraceExporter(observability.ContextWithExporterName(ctx, "zipkin"), len(td.Spans), len(td.Spans)-goodSpans)
+	body, err := ze.serializer.Serialize(tbatch)
+	if err != nil {
+		return len(td.Spans), err
+	}
 
-	return nil
+	req, err := http.NewRequest("POST", ze.url, bytes.NewReader(body))
+	if err != nil {
+		return len(td.Spans), err
+	}
+	req.Header.Set("Content-Type", ze.serializer.ContentType())
```

Comment:
@pjanotti , is this the place you mentioned about setting "content-type"?

Reply:
Exactly!

Reply:
Shall I consider this as resolved? Thanks.

Reply:
Yes, resolving.